test/cpp/src/TestClient.cpp

```diff
@@ -488,7 +488,7 @@ int main(int argc, char** argv) {
       testClient.testBinary(bin_result, string(bin_data, 256));
       if (bin_result.size() != 256) {
         printf("}\n*** FAILED ***\n");
-        printf("invalid length: %lu\n", bin_result.size());
+        printf("invalid length: %zu\n", bin_result.size());
```

Comment:
Given this is in test code, why not use something that is more flexible:
```printf(boost::str(boost::format("invalid length: %1%\n") % bin_result.size()));```
instead?